core/profile.cpp

```diff
@@ -673,7 +673,7 @@ static void setup_gas_sensor_pressure(const struct dive *dive, const struct dive
 
 	/* FIXME: The planner uses a dummy one-past-end cylinder for surface air! */
 	int num_cyl = pi.nr_cylinders + 1;
-	std::vector<int> seen(num_cyl, 0);
+	std::vector<bool> seen(num_cyl, false);
```

Comment:
Please don't use `std::vector<bool>`. This has some insane optimization, which usually is a pessimization. A huge error in the stdlib.

Reply:
Ha, I didn't know. Yes, using bitfields for this seems to be an odd choice for a high level language. Fixed.

Reply:
Interesting that they are using bitfields here - seems to be an odd choice of implementation for a high-level language. Fixed.